### DISCUSSION TITLE
请问我使用jsx怎么接收不到分析组件，作用域插槽的值，模板语法可以（是我jsx写的有问题嘛），求教大佬

### DISCUSSION THREAD

**Discussion by whyds000** [zouyaoji/vue-cesium#265] [Q&A] [answered]
**jsx代码**
`import { defineComponent, ref } from "vue";
import styles from './ps.module.less'
interface _emit{
  (e:'closePs'):void
}
export default defineComponent({
  setup(props,{emit}:{emit:_emit}){
    const ps=ref<any>()
    const editable=ref<boolean>(true)
    const scope=ref<any>()
    function clear(){
      ps.value.clearAll()
    }
    function closeEvt(){
      emit('closePs')
  
    }
    const slots={
     body:(scope:any)=>{
      scope.map((item:any,index:number)=>{
        return <vc-btn key={index}>{item.tip}</vc-btn>
      })
     }
    }
    return () => (  
      <div class={styles['ps-container']}>
         <span class={styles['delete']}  onClick={closeEvt}>x</span>
         <vc-analysis v-slots={slots}  editable={editable.value} position='bottom-left' ref={ps.value} offset={[10,30]}>
         </vc-analysis>
      </div>
    );
  }
})
`
**模板语法**
`<template>
  <div class="p-container">
    <span class="delete" @click="closeEvt">x</span>
    <vc-analyses :editable="editable" position="bottom-left" ref="ps" :offset="[10, 30]">
      <template #body="scope">
        <vc-btn
          class="v-btn"
          v-for="(item, index) in scope"
          :key="index"
          @click="clickEvt(item)"
          :color="item.isActive ? 'amber' : 'primary'"
          >{{ item.tip }}</vc-btn
        >
        <vc-btn class="v-btn edit" @click="editEvt">编辑 </vc-btn>
        <vc-btn class="v-btn" rounded color="red" @click="clear">清除</vc-btn>
      </template>
    </vc-analyses>
  </div>
</template>`

**Comment by zouyaoji** [comment, marked as answer]
```tsx
/*
 * @Author: zouyaoji@https://github.com/zouyaoji
 * @Date: 2022-08-23 19:08:12
 * @LastEditTime: 2022-08-23 19:21:06
 * @LastEditors: zouyaoji
 * @Description:
 *
 * @FilePath: \vue-cesium-demo\src\components\tsx-test\index.tsx
 */
import { defineComponent, ref } from 'vue'
import { VcAnalyses, VcAnalysesRef, VcBtn } from 'vue-cesium'
import { VcDrawingActionInstance } from 'vue-cesium/es/utils/drawing-types'

export default defineComponent({
  name: 'Test',
  setup(props, ctx) {
    const analysisRef = ref<VcAnalysesRef>(null)

    const toggle = (drawingActionInstance: VcDrawingActionInstance) => {
      analysisRef.value.toggleAction(drawingActionInstance.name)
    }
    const slots = {
      body: (drawingActionInstances: Array<VcDrawingActionInstance>) => {
        const child = []
        drawingActionInstances.forEach(drawingActionInstance => {
          child.push(
            <VcBtn
              rounded
              onClick={() => toggle(drawingActionInstance)}
              color={drawingActionInstance.isActive ? 'positive' : 'primary'}>
              {drawingActionInstance.name}
            </VcBtn>
          )
        })

        return <div class='custom-analysis'>{child}</div>
      }
    }
    return () => [<VcAnalyses v-slots={slots} ref={analysisRef}></VcAnalyses>]
  }
})

```


![image](https://user-images.githubusercontent.com/24430116/186145809-f0bcdd93-57c0-4262-a22b-d3d0d91bae40.png)

**Reply by whyds000** [reply]
感谢大佬，成功运行
